src/source/Crypto/Crypto.h

```diff
@@ -8,10 +8,32 @@ extern "C" {
 #endif
 
 #ifdef KVS_USE_OPENSSL
-#define KVS_RSA_F4                  RSA_F4
-#define KVS_MD5_DIGEST_LENGTH       MD5_DIGEST_LENGTH
-#define KVS_SHA1_DIGEST_LENGTH      SHA_DIGEST_LENGTH
+#define KVS_RSA_F4             RSA_F4
+#define KVS_MD5_DIGEST_LENGTH  MD5_DIGEST_LENGTH
+#define KVS_SHA1_DIGEST_LENGTH SHA_DIGEST_LENGTH
+
+// https://www.openssl.org/docs/man3.1/man3/MD5.html
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
+#define KVS_MD5_DIGEST(m, mlen, ob)                                                                                                                  \
+    do {                                                                                                                                             \
+        EVP_MD_CTX* mdctx;                                                                                                                           \
+        const EVP_MD* md;                                                                                                                            \
+        md = EVP_MD_fetch(NULL, "MD5", NULL);                                                                                                        \
+        if (md == NULL) {                                                                                                                            \
+            DLOGI("Failed to fetch MD5 provider");                                                                                                   \
+        } else {                                                                                                                                     \
+            mdctx = EVP_MD_CTX_new();                                                                                                                \
+            EVP_DigestInit_ex(mdctx, md, NULL);                                                                                                      \
+            EVP_DigestUpdate(mdctx, m, mlen);                                                                                                        \
+            EVP_DigestFinal_ex(mdctx, ob, NULL);                                                                                                     \
+            EVP_MD_CTX_free(mdctx);                                                                                                                  \
+            EVP_MD_free((EVP_MD*) md);                                                                                                               \
+        }                                                                                                                                            \
```

Comment:
I see in the docs it says:

>Ignoring failure returns of EVP_DigestInit_ex(), EVP_DigestInit_ex2(), or EVP_DigestInit() can lead to undefined behavior on subsequent calls updating or finalizing the EVP_MD_CTX such as the EVP_DigestUpdate() or EVP_DigestFinal() functions. The only valid calls on the EVP_MD_CTX when initialization fails are calls that attempt another initialization of the context or release the context.

There is an example here:
```
mdctx = EVP_MD_CTX_new();
    if (!EVP_DigestInit_ex2(mdctx, md, NULL)) {
        printf("Message digest initialization failed.\n");
        EVP_MD_CTX_free(mdctx);
        exit(1);
    }
    if (!EVP_DigestUpdate(mdctx, mess1, strlen(mess1))) {
        printf("Message digest update failed.\n");
        EVP_MD_CTX_free(mdctx);
        exit(1);
    }
    if (!EVP_DigestUpdate(mdctx, mess2, strlen(mess2))) {
        printf("Message digest update failed.\n");
        EVP_MD_CTX_free(mdctx);
        exit(1);
    }
    if (!EVP_DigestFinal_ex(mdctx, md_value, &md_len)) {
        printf("Message digest finalization failed.\n");
        EVP_MD_CTX_free(mdctx);
        exit(1);
    }
    EVP_MD_CTX_free(mdctx);
```

Taken from here:  https://www.openssl.org/docs/man3.0/man3/EVP_DigestInit_ex2.html

We should confirm if we need to call `EVP_DigestInit_ex` or `EVP_DigestInit_ex2`.  Regardless the error checking applies to both.

Reply:
It looks like this was still not addressed?

Reply:
I missed removing this macro altogether. Done. Also  about `EVP_DigestInit_ex` vs `EVP_DigestInit_ex2`. ex2 takes in a params options that we can use to configure the hashing operation, but we are not doing that. So `EVP_DigestInit_ex` is ok for now.